### PR TITLE
Blittable bools, VS2010 compatibility

### LINKED ISSUE #5: Ensure structs are blittable
Apparently bool needs special care in C# and needs something like `[MarshalAs(UnmanagedType.U1)]` so that .NET Framework doesn't get mad about being a nonblittable type. SDL_Event in particular is affected by this; we need to cast to `SDL.SDL_Event*` in a couple places and there's no real way around this.

CC @kg

### PR DESCRIPTION
**Pull request by thatcosmonaut**
Resolves #5

### CODE REVIEW

File: SDL3/SDL3.cs
Comment (flibitijibibo) [inline SDL3/SDL3.cs:1]: The diff here seems unusually large... could it have flipped to DOS line endings maybe?

**Review by flibitijibibo — APPROVED** [review]
As long as it passes @kg's .NET tests this should be okay.

File: SDL3/SDL3.cs
Comment (thatcosmonaut) [reply SDL3/SDL3.cs:1]: I actually ran the generator on my Mac... I wonder if the function being added at the top + flipping all the bytes to SDLBool threw off the git diff

Comment (thatcosmonaut) [reply SDL3/SDL3.cs:1]: Re-ran on Windows and no change, so I think it's just git being silly.

**Comment by thatcosmonaut** [comment]
Running dos2unix fixed the insane diff, just need a re-review.

File: GenerateBindings/Program.cs [hunk 1]
```diff
@@ -704,6 +704,50 @@ private static string DecodeFromUTF8(IntPtr ptr, bool shouldFree = false)
...
+
+        public static implicit operator bool(SDLBool b) => b.value != FALSE_VALUE;
+
+        public static implicit operator SDLBool(bool b) => new SDLBool(b ? TRUE_VALUE : FALSE_VALUE);
+
+        public bool Equals(SDLBool other) => (bool)other == (bool)this;
+
+        public override bool Equals(object rhs)
+        {{
+            if (rhs is bool rhsBool)
```
Comment (kg) [inline GenerateBindings/Program.cs:734]: i'm not sure this convenience cast syntax will build in vs2010 C#

**Comment by kg** [comment]
Current rev of SDL3.cs builds unmodified for me on VS2022+netfx and gets far enough for FNA3D to crash, so I think you've got it. Not sure about VS2010

**Review by flibitijibibo — APPROVED** [review]
CI's happy so I'm happy!

**Pull request merged**